assets/src/components/SurveyModal.js

```diff
@@ -50,8 +51,25 @@ const useStyles = makeStyles((theme) => ({
 
 export default function SurveyModal (props) {
   const classes = useStyles()
+  const location = useLocation()
   const [open, setOpen] = useState(false)
 
+  const lastTwoSteps = location.pathname.split('/').slice(-2)
+  const viewName = lastTwoSteps.length < 2
+    ? ''
+    : lastTwoSteps[0] === 'courses'
+      ? 'courses'
+      : lastTwoSteps[1]
+
+  const params = {
+    userID: props.user.LTIlaunchID,
+    userName: props.user.username,
+    courseID: props.courseID,
+    view: viewName
+  }
+  const searchParams = new URLSearchParams()
+  Object.entries(params).map(([key, value]) => searchParams.append(key, value))
```

Comment:
Very nice.  I'd hoped there would be something like this available.  I didn't really like my initial approach, concatenating a string myself.